### PR TITLE
Provide an example for DataPostprocessorVector.

### PR DESCRIPTION
**Pull request by bangerth**

### CODE REVIEW

File: include/deal.II/numerics/data_postprocessor.h [hunk 1]
```diff
@@ -591,8 +595,169 @@ class DataPostprocessorScalar : public DataPostprocessor<dim>
...
+ *                      computed_quantities.size());
+ *
+ *     for (unsigned int p=0; p<input_data.solution_gradients.size(); ++p)
+ *       {
+ *         AssertDimension (computed_quantities[p].size(), dim);
+ *         for (unsigned int d=0; d<dim; ++d)
+ *           // like above, but also multiply the gradients with
+ *           // the coefficient evaluated at the current point:
+ *           computed_quantities[p][d]
+ *             = input_data.solution_gradients[p][d];
```
Comment (jppelteret) [inline include/deal.II/numerics/data_postprocessor.h]: This doesn't seem to match the comment above. Are you missing the coefficient here?

File: include/deal.II/numerics/data_postprocessor.h [hunk 2]
```diff
@@ -591,8 +595,169 @@ class DataPostprocessorScalar : public DataPostprocessor<dim>
...
+ * In the second image, the background color corresponds to the magnitude of the
+ * gradient vector and the vector glyphs to the gradient itself. It may be surprising
+ * at first to see that from each vertex, multiple vectors originate, going in
+ * different directions. But that is because the solution is only continuous:
+ * in general, the gradient is discontinuous across edges, and so the multiple
+ * vectors originating from each vertex simply represent the differing
+ * gradients of the solution at each adjacent cell.
+ *
+ * The output above -- namely, the gradient $\nabla u$ of the solution --
+ * corresponds to the "strain" in linear elasticity if the solution were scalar.
```
Comment (jppelteret) [inline include/deal.II/numerics/data_postprocessor.h]: I think that relating this to linear elasticity is more confusing than it is helpful. Can't we rather use a physical scalar field, e.g. temperature, that offers a much more intuitive understanding of the results and later discussion?

File: include/deal.II/numerics/data_postprocessor.h [hunk 3]
```diff
@@ -591,8 +595,169 @@ class DataPostprocessorScalar : public DataPostprocessor<dim>
...
+ * It is very small in the central part of the domain because in step-6 we are
+ * solving an equation that has a coefficient $a(\mathbf x)$ that is large in
+ * the central part and small on the outside. On the other hand, the "stress"
+ * corresponds to the quantity $a(\mathbf x) \nabla u(\mathbf x)$. For the
```
Comment (jppelteret) [inline include/deal.II/numerics/data_postprocessor.h]: Vector valued stress? I can see this being really confusing! If we talk temperature gradients, heat flux and conduction coefficients then we won't lose the engineers along the way :-P

File: include/deal.II/numerics/data_postprocessor.h [hunk 4]
```diff
@@ -591,8 +595,169 @@ class DataPostprocessorScalar : public DataPostprocessor<dim>
...
+ * a continuous function. There are (large) vectors around the interface
+ * where the coefficient jumps (at half the distance between the center of the
+ * disk to the perimeter) that seem to point in the wrong direction; this is
+ * an artifact of the fact that the solution has a discontinuous gradient
+ * at these points and that the numerical solution on the current grid
+ * does not adequately resolve this interface. This, however, is not
+ * important to the current discussion.
+ *
+ *
+ * <h3> Extension to the real "stress" </h3>
```
Comment (jppelteret) [inline include/deal.II/numerics/data_postprocessor.h]: If we do change "stress" to "heat flux" in the above discussion, then I think we could just change this title to something like "Extension to tensor valued fields" and keep what you've got written below. The only thing we'd need to change is "real stress" to "stress".

File: include/deal.II/numerics/data_postprocessor.h [hunk 5]
```diff
@@ -591,8 +595,169 @@ class DataPostprocessorScalar : public DataPostprocessor<dim>
...
+ *
+ * The example above uses a scalar solution and its gradient as an example.
+ * On the other hand, one may want to do something similar for the real
+ * stress of a vector-valued displacement field (such as those computed in
+ * step-8, step-7, or step-44). In that case, the solution is already vector
```
Comment (jppelteret) [inline include/deal.II/numerics/data_postprocessor.h]: step-18 as well?

File: include/deal.II/numerics/data_postprocessor.h (hunk 1 above)
Comment (bangerth) [reply include/deal.II/numerics/data_postprocessor.h]: Yes, good spot!

**Comment by bangerth** [comment]
Excellent points, @jppelteret . I have no idea why I didn't think of heat flux right away. I've made the changes to phrase everything in terms of temperature and heat flux.

We don't currently support tensorial output. But I think it would still be nice to have the equivalent of these classes, say `DataPostprocessorTensor` and `DataPostprocessorSymmetricTensor`. They would output individual tensor components as scalars, so maybe that's not quite that useful, but it would still make some things easier. What do you think?

**Comment by tjhei** [comment]
@bangerth Can you please also add this example as a test and leave a note there that this is the test for the example in the documentation? I don't care if we test the specific output, but this will remind us to update the example in case we change something in the interface in the future and makes it easy to test it if needed.

**Comment by bangerth** [comment]
@tjhei -- good suggestion. I added the test.

**Comment by jppelteret** [comment]
> We don't currently support tensorial output. But I think it would still be nice to have the equivalent of these classes, say DataPostprocessorTensor and DataPostprocessorSymmetricTensor. They would output individual tensor components as scalars, so maybe that's not quite that useful, but it would still make some things easier. What do you think?

I think that this is a great idea, and when (if) we ever implement a tensorial output to vtu/vtk then we could update these classes. Although I don't have the time to implement these classes right now, I don't mind doing so in the future. I've opened issue #4995 in this respect.

**Pull request merged**